### PR TITLE
[Merged by Bors] - feat: port Algebra.Parity

### PR DESCRIPTION
**Pull request by siddhartha-gadgil**
dcf2250875895376a142faeeac5eabff32c48655

- [x] depends on: #1055

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1055
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1055~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Algebra/Parity.lean [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+--   [to_additive Even.exists_two_nsmul
+--       "Alias of the forwards direction of\n`even_iff_exists_two_nsmul`."]
+--   IsSquare.exists_sq
+
+-- attribute
+--   [to_additive even_of_exists_two_nsmul
+--       "Alias of the backwards direction of\n`even_iff_exists_two_nsmul`."]
+--   is_square_of_exists_sq
+
+-- @[to_additive Even.nsmul]
```
Comment (ChrisHughes24) [inline Mathlib/Algebra/Parity.lean:107]: Why were these commented out?

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**